### ISSUE TITLE
binary_crossentropy() usage example is misleading

### ISSUE THREAD

**Issue by johnmaccormick** [tensorflow/tensorflow#50828]
## URL(s) with the issue:
https://www.tensorflow.org/versions/r2.5/api_docs/python/tf/keras/losses/binary_crossentropy

The same problem occurs across many versions and aliases of this function.

## Description of issue (what needs changing):
The "standalone usage" example appears to be misleading, because it implies that rows and/or columns of the `y_pred` parameter should sum to 1.0. 

As I am new to TensorFlow, I am a little apprehensive about pointing out a potential problem in the documentation and I will be very grateful if an experienced practitioner can confirm whether my suggestion is correct or alternatively if I have misunderstood something. Thank you!

### Usage example

In the usage example given here, the `y_pred` parameter is `[[0.6, 0.4], [0.4, 0.6]]`. Any reader with a background in probability and statistics will immediately assume that these numbers represent normalized probabilities, because the rows and the columns sum to 1.0. Furthermore, because the name of the method is *binary* cross entropy, a reader is likely to assume that the rows or columns of y_pred represent the two possible binary outcomes. In fact, none of these assumptions is correct. The parameter `y_pred` can be any shape (subject to a certain restriction mentioned below). Each entry in `y_pred` has a value that does not depend on the other values in the tensor. Example code demonstrating this is attached.

My feeling is that a better usage example would use numbers that do not sum to 1.0, so that readers will not make the mistaken assumption that they represent normalized probabilities. In addition, the usage example could be further improved by breaking the symmetry of the 2x2 example, and avoiding using two rows or two columns so that it is clear the "binary" nature of the cross entropy has nothing to do with the number of rows and columns here. The smallest example conveying all of these ideas would probably be 4x3, as in the attached code.

Technical note: There is a possibility of confusion when we say that the parameter does not represent normalized probabilities. This is certainly true when the parameter `from_logits` is `True`. However, when `from_logits` is `False`, each element in `y_pred` does in fact represent a probability, and as with any other true probability it is a normalized probability. However, it is not normalized with respect to any other elements of the tensor. For example, the value 0.8 represents a prediction that the an instance has an 80% chance of being a positive instance.  This means there is a 20% chance of that same instance being negative, but the value 0.2 does not appear anywhere in the `y_pred` tensor; the 0.2 is implicit in the 0.8.

Details about the shape for `y_pred`: it must have either the same shape as `y_true`, or a shape that is compatible with the shape of `y_true` after broadcasting has been applied to `y_true`.

### Submit a pull request?

I will be happy to prepare and submit a pull request if an experienced contributor can confirm to me that my description of the problem is correct.

Again, I will be grateful if someone can let me know if my suggestions are correct and I do apologize in advance if I misunderstood something.

~~~~
import numpy as np
import tensorflow as tf


def my_sigmoid(x: np.ndarray):
    return 1 / (1 + np.exp(-x))


def safe_log(x: np.ndarray):
    eps = 1e-11
    return np.log(np.maximum(x, eps))


def my_cross_entropy(y: np.ndarray, z: np.ndarray, from_logits=True):
    if from_logits:
        z = my_sigmoid(z)
    return -(y * safe_log(z) + (1 - y) * safe_log((1 - z)))


def tf_cross_entropy(y: np.ndarray, z: np.ndarray, from_logits=True):
    y = tf.constant(y, dtype=tf.float32)
    z = tf.constant(z, dtype=tf.float32)
    return tf.keras.losses.binary_crossentropy(y_true=y, y_pred=z,
                                               from_logits=from_logits)


def compare_cross_entropy(y, z, from_logits=True):
    my_val = my_cross_entropy(y, z, from_logits)
    tf_val = tf_cross_entropy(y, z, from_logits)
    print(f"y {y}\nz {z}\n")
    print(f"my_val {my_val}\ntf_val {tf_val}\n")
    print(f"my_val shape {my_val.shape}\ntf_val shape {tf_val.shape}\n")


##########
# main ###
##########
y_val = np.array([[1, 1, 1],
                  [1, 1, 1],
                  [0, 0, 0],
                  [0, 1, 0], ], dtype=np.float32)

z_val = np.array([[0.4, 0.45, 0.46],
                  [0.9, 0.95, 0.96],
                  [0.0, 0.0, 0.0],
                  [0.7, 0.7, 0.7],
                  ], dtype=np.float32)

print(f'y shape {y_val.shape}, z shape {z_val.shape}')
compare_cross_entropy(y_val, z_val, from_logits=False)
~~~~

**Comment by tilakrayal** [comment]
@johnmaccormick ,
Please post this issue on [keras-team/keras](https://github.com/keras-team/keras/issues) repo.
To know more refer to:
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)

**Comment by johnmaccormick** [comment]
Thanks, this has been reposted as requested:

https://github.com/keras-team/keras/issues/14955

**Comment by tilakrayal** [comment]
@johnmaccormick ,

Please feel free to move this issue to closed status since it is already being tracked in Keras repo.Thanks!
